### PR TITLE
More test abstractions for multiple databases

### PR DESCRIPTION
**Pull request by sohkai**
Removes some leftover coupling in the tests with RethinkDB.

**[Proposal]**: I've moved anything RethinkDB-specific into its own file or folder under the higher-level logical grouping of tests, rather than moving them all into their own folder. I'm not in love with how this is organized, but because of the issues surrounding the `tests/db/` folder, I think this is the best way to structure the tests for now.

**Todo**:
- [x] Skip tests unrelated to the currently selected database (ended up going with a pytest hook)
- [x] Move `tests/db/rethinkdb/test_schema.py` to `tests/backends/rethinkdb`

### CODE REVIEW

File: tests/commands/rethinkdb/test_commands.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import pytest
```
Comment (sohkai) [inline tests/commands/rethinkdb/test_commands.py]: RethinkDB-specific tests moved from the old `tests/test_commands.py`.

File: tests/commands/conftest.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import pytest
```
Comment (sohkai) [inline tests/commands/conftest.py]: Fixtures moved from the old `tests/test_commands.py`.

**Comment by r-marques** [comment]
Regarding your todo item http://doc.pytest.org/en/latest/example/pythoncollection.html#customizing-test-collection-to-find-all-py-files

Assuming the tests are well organized, it should be easy to just add to `collect_ignore` the folders we want to skip

**Comment by r-marques** [comment]
There is also `tests/db/rethinkdb/test_schema.py` that should be moved to `tests/backends/rethinkdb`

**Pull request merged**